dags/backfill.py

```diff
@@ -0,0 +1,130 @@
+    https://airflow.apache.org/docs/apache-airflow/stable/concepts/params.html
+    """
+    params_parsed = ast.literal_eval(string_params)
+    return BackfillParams(**params_parsed)
+
+
+def dry_run_branch_callable(params: str) -> str:
+    backfill_params = __parse_string_params(params)
+    return "dry_run" if backfill_params.dry_run else "real_deal"
+
+
+def clear_branch_callable(params: str) -> str:
+    backfill_params = __parse_string_params(params)
+    return "clear_tasks" if backfill_params.clear else "do_not_clear_tasks"
+
+
+def param_validation(params: str) -> bool:
+    backfill_params = __parse_string_params(params)
+    backfill_params.validate_date_range()
+    validate_dag_exists(dag_name=backfill_params.dag_name)
+    backfill_params.validate_regex_pattern()
+    return True
+
+
+def validate_dag_exists(dag_name: str) -> None:
+    dag_instance = DagModel.get_dagmodel(dag_name)
+    if dag_instance is None:
+        raise ValueError(f"`dag_name`={dag_name} does not exist")
+
+
+def generate_bash_command(params: str) -> str:
+    backfill_params = __parse_string_params(params)
+    return " ".join(backfill_params.generate_backfill_command())
+
+
+doc_md = """
+# Backfill DAG
+
+#### Use with caution 
+
+#### Some tips/notes:
+
+* Always use dry run first. Especially when using task regex
+* Date formats are 2020-03-01 or 2020-03-01T00:00:00
+* Dry run for clearing tasks will show you the list of tasks that will be cleared
+* Dry run for backfilling will not show the list, but is useful in testing for input errors
+
+"""
+
+
+@dag(
+    dag_id='backfill',
+    schedule_interval=None,
+    doc_md=doc_md,
+    catchup=False,
+    start_date=datetime.datetime(2022, 11, 1),
+    dagrun_timeout=datetime.timedelta(days=1),
+    tags=[Tag.ImpactTier.tier_3],
+    params={"dag_name": "dag_name",
+            "start_date": (datetime.date.today() - datetime.timedelta(days=10)).isoformat(),
+            "end_date": datetime.date.today().isoformat(),
+            "clear": False,
+            "dry_run": True,
+            "task_regex": None,
+            }
+)
+def backfill_dag():
+    param_validation_task = PythonOperator(
+        task_id="param_validation",
+        python_callable=param_validation,
+        op_kwargs={"params": "{{ dag_run.conf }}"},
+    )
+
+    dry_run_branch_task = BranchPythonOperator(
+        task_id="dry_run_parameter",
+        python_callable=dry_run_branch_callable,
+        op_kwargs={"params": "{{ dag_run.conf }}"},
+        trigger_rule=TriggerRule.ONE_SUCCESS,
+    )
+
+    dry_run_task = DummyOperator(task_id="dry_run")
+    real_deal_task = DummyOperator(task_id="real_deal")
+
+    clear_branch_task = BranchPythonOperator(
+        task_id="clear_parameter",
+        python_callable=clear_branch_callable,
+        op_kwargs={"params": "{{ dag_run.conf }}"},
+        trigger_rule=TriggerRule.ONE_SUCCESS,
+    )
+
+    clear_tasks_task = DummyOperator(task_id="clear_tasks")
```

Comment:
Just curious, why do we use DummyOperators here?

Reply:
Just for visual representation. This way you can see at a glance if you're doing a dry run and clearing tasks. The parent task is a branch operator, and the dummy tasks displays the parameter value.